### PR TITLE
Refine CloudTrailLake for S3PresignUrl

### LINKED ISSUE #76: Create a mockup/trial model for how CloudTrail accesses can be rendered inside Elsa
The CloudTrail logs are able to be queried looking for a subset of entries related to each 'release'.

Whilst cloudtrail is excellent for actual audit (like a legal investigation) - the data is probably better situated inside Elsa where it is accessible directly from users/data custodians (who in general won't have access to the AWS backend)

We need to decide on what data we want to bring across (all? some? summarised?) into the Elsa Data db and what level of functionality we want to provide to users.

Some considerations
- log size? We don't want 5 Gib of cloud trail data causing other aspects of Elsa Data to be slow/inefficient..
- UI functionality - what is actually going to be useful to spot outlier behaviour?
- can we make a model that works with all clouds - not AWS specific..

This ticket is just about getting some early thinking - and perhaps a UI mockup of what might be possible to stimulate some thinking..

### PR DESCRIPTION
**Pull request by williamputraintan**
- Complete mechanism to fetch real data from AWS eventDataStore
- Possible putting this on cron job but could trigger this on-demand by the api layer (`/api/releases/:rid/access-log/sync`). A button is put on the UI.
	Screen Shot 2022-12-08 at 4 48 04 pm
- Add `lastDateTimeDataAccessLogQuery` to release property. To keep track last querying date.

### CODE REVIEW

File: application/backend/src/api/routes/audit-log-routes.ts [hunk 1]
```diff
@@ -171,4 +175,73 @@ export const auditLogRoutes = async (fastify: FastifyInstance, _opts: any) => {
...
+    },
+    async function (request, reply) {
+      const { authenticatedUser } = authenticatedRouteOnEntryHelper(request);
+
+      const releaseId = request.params.rid;
+      const accessType = request.body.accessType;
+
+      let replyMessage = "";
+
+      switch (accessType) {
```
Comment (williamputraintan) [inline application/backend/src/api/routes/audit-log-routes.ts:212]: Just realising when making this PR. We might not need to distinguish between presign/access-point/htsget/etc. I suppose might just run all method (presign/access-point/etc) to query from cloudtrail?

I guess could implement this on later PR when introducing access point data tracking.

Comment (williamputraintan) [reply application/backend/src/api/routes/audit-log-routes.ts:212]: But I think still need to differentiate between cloud provider?

File: application/backend/src/api/routes/audit-log-routes.ts [hunk 2]
```diff
@@ -14,13 +14,15 @@ import {
...
+import { AwsCloudTrailLakeService } from "../../business/services/aws-cloudtrail-lake-service";
+import { DatasetService } from "../../business/services/dataset-service";
 
 export const AuditEventForReleaseQuerySchema = Type.Object({
   page: Type.Optional(Type.Number()),
   orderByProperty: Type.Optional(Type.String()),
   orderAscending: Type.Optional(Type.Boolean()),
 });
-export type AuditEventForReleastQueryType = Static<
+export type AuditEventForReleasQueryType = Static<
```
Comment (andrewpatto) [inline application/backend/src/api/routes/audit-log-routes.ts]: this is Releas but probably should be Release

File: application/backend/src/api/routes/audit-log-routes.ts (hunk 1 above)
Comment (andrewpatto) [reply application/backend/src/api/routes/audit-log-routes.ts:212]: Good catch - but yes wouldn't worry about it for now.. can deal with when we get to our first "other" cloud

File: application/backend/src/api/routes/audit-log-routes.ts (hunk 2 above)
Comment (williamputraintan) [reply application/backend/src/api/routes/audit-log-routes.ts]: Yup, fixing it!

File: application/backend/config/datasets.json5 [hunk 3]
```diff
@@ -14,6 +14,9 @@
       uri: "urn:fdc:umccr.org:2022:dataset/10g",
       storageLocation: "aws-s3",
       storageUriPrefix: "s3://umccr-10g-data-dev",
+      aws: {
+        eventDataStoreId: "327383f8-3273-3273-3273-327383f8fc43",
```
Comment (andrewpatto) [inline application/backend/config/datasets.json5:18]: So the definition of the event data store is per dataset (which makes sense) - but I presume for instance in AG we will just have a single event store defined - and every dataset will point at the same one?

**Review by andrewpatto — APPROVED** [review]

File: application/backend/config/datasets.json5 (hunk 3 above)
Comment (williamputraintan) [reply application/backend/config/datasets.json5:18]: Yes I think is likely and possible.

Comment (williamputraintan) [reply application/backend/config/datasets.json5:18]: Well I haven't thought if a dataset could be stored in multiple location (hopefully not 😅)?

**Pull request merged**